packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -941,68 +1146,777 @@ describe('Plugin Setup Processor', function () {
+        expect(
+          await PluginV1.attach(proxy).callStatic.getImplementationAddress()
+        ).to.equal(await setupV1.callStatic.getImplementationAddress());
+      });
+
+      it('initializes the members', async () => {
+        expect(await PluginV1.attach(proxy).v1()).to.equal(1);
+      });
+
+      it('sets the V1 helpers', async () => {
+        expect(helpersV1).to.deep.equal(mockHelpers(1));
+      });
 
-      // TODO: Find a way to test upgradeProxy
-      // also check this function's errors as they might be misleading
-      // it also get thrown if UPGRADE_PLUGIN_PERMISSION_ID permission is not granted
-      // it('applyUpdate: reverts if PluginNonupgradeable', async () => {});
+      it('sets the V1 permissions', async () => {
+        expect(permissionsV1).to.deep.equal(
+          mockPermissionsOperations(0, 1, Operation.Grant).map(perm =>
+            Object.values(perm)
+          )
+        );
+      });
+
+      it('updates to V2', async () => {
+        await updateHelper(
+          psp,
+          targetDao,
+          proxy,
+          repoUUPS,
+          helpersV1,
+          setupV1,
+          setupV2
+        );
+      });
 
-      it('correctly applies an update', async () => {
-        const daoAddress = targetDao.address;
-        const pluginSetupV1 = pluginSetupV1Mock.address;
+      it('updates to V3', async () => {
+        await updateHelper(
+          psp,
+          targetDao,
+          proxy,
+          repoUUPS,
+          helpersV1,
+          setupV1,
+          setupV3
+        );
+      });
 
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
+      it('cannot update again to V1', async () => {
+        await expect(
+          updateHelper(
             psp,
-            daoAddress,
-            pluginSetupV1,
-            pluginSetupMockRepoAddress,
-            EMPTY_DATA
+            targetDao,
+            proxy,
+            repoUUPS,
+            helpersV1,
+            setupV1,
+            setupV1
+          )
+        ).to.be.reverted;
+      });
+
+      context(`and updated to V2`, function () {
+        let helpersV2: string[];
+        let permissionsV1V2: PermissionOperation[];
+        let initDataV1V2: BytesLike;
+
+        beforeEach(async () => {
+          ({
+            updatedHelpers: helpersV2,
+            permissions: permissionsV1V2,
+            initData: initDataV1V2,
+          } = await updateHelper(
+            psp,
+            targetDao,
+            proxy,
+            repoUUPS,
+            helpersV1,
+            setupV1,
+            setupV2
+          ));
+        });
+
+        /*it('cannot update to V2 again', async () => {
+          await expect(
+            updateHelper(
+              psp,
+              targetDao,
+              proxy,
+              pluginRepo,
+              helpersV2,
+              setupV2,
+              setupV2
+            )
+          ).to.be.reverted;
+        });*/
```

Comment:
why it is commented out?

Reply:
Good question... 😅 I removed the comments markers